File: sdks/java/src/main/java/com/puxbay/resources/BranchesResource.java
package com.puxbay.resources;

import com.puxbay.Puxbay;
import com.puxbay.exceptions.PuxbayException;
import com.puxbay.models.PaginatedResponse;
import com.puxbay.models.Branch;

public class BranchesResource {
    private final Puxbay client;

    public BranchesResource(Puxbay client) {
        this.client = client;
    }

    public PaginatedResponse<Branch> list(int page) throws PuxbayException {
        return (PaginatedResponse<Branch>) client.request("GET", "branches/?page=" + page, null, PaginatedResponse.class);
    }

    public Branch get(String branchId) throws PuxbayException {
        return client.request("GET", "branches/" + branchId + "/", null, Branch.class);
    }

    public Branch create(Branch branch) throws PuxbayException {
        return client.request("POST", "branches/", branch, Branch.class);
    }

    public Branch update(String branchId, Branch branch) throws PuxbayException {
        return client.request("PATCH", "branches/" + branchId + "/", branch, Branch.class);
    }

    public void delete(String branchId) throws PuxbayException {
        client.request("DELETE", "branches/" + branchId + "/", null, Void.class);
    }
}

File: sdks/kotlin/src/main/kotlin/com/puxbay/resources/BranchesResource.kt
package com.puxbay.resources

import com.puxbay.Puxbay
import com.puxbay.models.Branch
import com.puxbay.models.PaginatedResponse
import io.ktor.http.HttpMethod

class BranchesResource(client: Puxbay) : BaseResource(client) {
    suspend fun list(page: Int = 1): PaginatedResponse<Branch> =
        client.request(HttpMethod.Get, "branches/?page=$page")
    
    suspend fun get(branchId: String): Branch =
        client.request(HttpMethod.Get, "branches/$branchId/")
    
    suspend fun create(branch: Branch): Branch =
        client.request(HttpMethod.Post, "branches/", branch)
    
    suspend fun update(branchId: String, branch: Branch): Branch =
        client.request(HttpMethod.Patch, "branches/$branchId/", branch)
    
    suspend fun delete(branchId: String) {
        client.request<Unit>(HttpMethod.Delete, "branches/$branchId/")
    }
}
